src/update-function/swap-array-control.ts

```diff
@@ -0,0 +1,24 @@
+import { SwapArrayControlAction } from '../actions';
+import { formArrayReducer } from '../array/reducer';
+import { FormArrayState } from '../state';
+import { ensureState } from './util';
+
+/**
+ * This update function takes a source index, a destination index, and returns a projection function
+ * that swaps the child controls at the source and destination indices in a form array state.
+ */
+export function swapArrayControl<TValue>(from: number, to: number): (state: FormArrayState<TValue>) => FormArrayState<TValue>;
```

Comment:
Please move the generic `TValue` parameter to the returned lambda. This improves the type inference experience.

Also rename parameters to `fromIndex` and `toIndex`.